script/format-sql

```diff
@@ -0,0 +1,109 @@
+  'HASH', 'HAVING', 'IF', 'IGNORE', 'IN', 'INNER', 'INTERSECT', 'INTERVAL',
+  'INTO', 'IS', 'JOIN', 'LATERAL', 'LEFT', 'LIKE', 'LIMIT', 'LOOKUP',
+  'MERGE', 'NATURAL', 'NEW', 'NO', 'NOT', 'NULL', 'NULLS', 'OF', 'ON', 'OR',
+  'ORDER', 'OUTER', 'OVER', 'PARTITION', 'PRECEDING', 'PROTO', 'RANGE',
+  'RECURSIVE', 'RESPECT', 'RIGHT', 'ROLLUP', 'ROWS', 'SELECT', 'SET', 'SOME',
+  'STRUCT', 'TABLESAMPLE', 'THEN', 'TO', 'TREAT', 'TRUE', 'UNBOUNDED',
+  'UNION', 'UNNEST', 'USING', 'WHEN', 'WHERE', 'WINDOW', 'WITH', 'WITHIN',
+];
+// These words get their own line followed by increased indent
+const reservedToplevelWords = [
+  // DDL
+  'ALTER TABLE IF EXISTS', 'ALTER TABLE', 'CREATE OR REPLACE TABLE',
+  'CREATE TABLE IF NOT EXISTS', 'CREATE TABLE', 'DROP TABLE', 'DROP VIEW',
+  // DML
+  'DELETE FROM', 'DELETE', 'INSERT INTO', 'INSERT', 'MERGE INTO', 'MERGE',
+  'UPDATE',
+  // SQL
+  'CROSS JOIN', 'FROM', 'FULL JOIN', 'FULL OUTER JOIN', 'GROUP BY', 'HAVING',
+  'INNER JOIN', 'INTERSECT', 'JOIN', 'LEFT JOIN', 'LEFT OUTER JOIN', 'LIMIT',
+  'ORDER BY', 'OUTER JOIN', 'PARTITION BY', 'RIGHT JOIN', 'RIGHT OUTER JOIN',
+  'ROWS BETWEEN', 'ROWS', 'SELECT AS STRUCT', 'SELECT', 'SET', 'UNION ALL',
+  'UNION', 'USING', 'VALUES', 'WHERE',
+];
+// These words start a new line at the current indent
+const reservedNewlineWords = ['AND', 'BETWEEN', 'ELSE', 'OR', 'WHEN', 'XOR'];
+
+// BigQuery Standard SQL and capitalizing keywords aren't directly supported
+// by the sql-formatter library, so we implement support for those here
+function format(file) {
+  const tokenizer = new Tokenizer({
+    reservedWords,
+    reservedToplevelWords,
+    reservedNewlineWords,
+    // eslint-disable-next-line quotes
+    stringTypes: [`''`, '""', '``'],
+    openParens: ['(', 'CASE', '['],
+    closeParens: [')', 'END', ']'],
+    indexedPlaceholderTypes: ['?'],
+    namedPlaceholderTypes: ['@'],
+    lineCommentTypes: ['#', '--'],
+  });
+  // add >> and << to OPERATOR_REGEX to prevent spaces being added in the middle
+  tokenizer.OPERATOR_REGEX = /^(<<|>>|!=|<>|==|<=|>=|!<|!>|\|\||::|->>|->|~~\*|~~|!~~\*|!~~|~\*|!~\*|!~|.)/;
+  const formatter = new Formatter({}, tokenizer);
+  // read query file
+  const query = fs.readFileSync(file, 'utf-8');
+  // capitalize keywords
+  formatter.tokens = tokenizer.tokenize(query).map((token) => {
+    if (token.type.startsWith('reserved')) {
+      return { type: token.type, value: token.value.toUpperCase() };
+    }
+    return token;
+  });
+  // format whitespace
+  const formatted = `${formatter.getFormattedQueryFromTokens().trim()}\n`;
+  const changed = query !== formatted;
+  // write to stdout or log change and overwrite file
+  if (file === '/dev/stdin') {
+    process.stdout.write(formatted);
+  } else if (changed) {
+    console.log(`reformatted ${file}`);
+    fs.writeFileSync(file, formatted);
+  }
+  return changed;
+}
+
+function listSqlFiles(file) {
```

Comment:
It might be simpler long run to use something like [`klaw-sync`](https://www.npmjs.com/package/klaw-sync) instead of writing your own fswalker.

Alternatively, making this script work only on passed files and using something like [Therapist](https://github.com/rehandalal/therapist) could also simplify things.

Reply:
exactly what i was looking for and couldn't find last night 😄